pkg/operator/controller/machineconfig/controller.go

```diff
@@ -80,228 +86,138 @@ func New(mgr ctrl.Manager) *MachineConfigReconciler {
 // service(ex: CRI-O) requires debugging to be enabled/disabled through the
 // MachineConfigs, controller creates the required MachineConfigs, MachineConfigPool
 // and labels the nodes where the changes are to be applied.
-func (r *MachineConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, err error) {
+func (r *MachineConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	if ctxLog, err := logr.FromContext(ctx); err == nil {
 		r.Log = ctxLog
 	}
 
 	r.Log.V(1).Info("reconciliation started")
 
 	// Fetch the NodeObservabilityMachineConfig CR
-	r.CtrlConfig = &v1alpha2.NodeObservabilityMachineConfig{}
-	if err = r.ClientGet(ctx, req.NamespacedName, r.CtrlConfig); err != nil {
-		if kerrors.IsNotFound(err) {
+	nomc := &v1alpha2.NodeObservabilityMachineConfig{}
+	if err := r.ClientGet(ctx, req.NamespacedName, nomc); err != nil {
+		if errors.IsNotFound(err) {
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
 			r.Log.V(1).Info("nodeobservabilitymachineconfig resource not found. Ignoring as it could have been deleted")
 			return ctrl.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
-		return ctrl.Result{RequeueAfter: defaultRequeueTime}, fmt.Errorf("failed to fetch nodeobservabilitymachineconfig: %w", err)
+		return ctrl.Result{}, fmt.Errorf("failed to fetch nodeobservabilitymachineconfig %q: %w", req.NamespacedName, err)
 	}
 
-	r.Log.V(1).Info("nodeobservabilitymachineconfig resource found", "nodeselector", r.CtrlConfig.Spec.NodeSelector)
+	r.Log.V(1).Info("reconciling", "nodeobservabilitymachineconfig", req.NamespacedName)
```

Comment:
I think that we can add the whole `NOB` resource, this will show the whole definition, can be useful. `req` doesn't make sense, it should be there already.

Reply:
I've removed this log, adding the entire NOB would just add too much noise for now.